apps/api/src/app/inbox/usecases/session/session.usecase.ts

```diff
@@ -0,0 +1,94 @@
+  SelectIntegrationCommand,
+  SelectIntegration,
+  AuthService,
+} from '@novu/application-generic';
+
+import { ApiException } from '../../../shared/exceptions/api.exception';
+import { SessionCommand } from './session.command';
+import { SessionResponseDto } from '../../dtos/session-response.dto';
+import { AnalyticsEventsEnum } from '../../utils';
+import { validateHmacEncryption } from '../../utils/encryption';
+import { NotificationsCount } from '../notifications-count/notifications-count.usecase';
+import { NotificationsCountCommand } from '../notifications-count/notifications-count.command';
+
+@Injectable()
+export class Session {
+  constructor(
+    private environmentRepository: EnvironmentRepository,
+    private createSubscriber: CreateSubscriber,
+    private authService: AuthService,
+    private selectIntegration: SelectIntegration,
+    private analyticsService: AnalyticsService,
+    private notificationsCount: NotificationsCount
+  ) {}
+
+  @LogDecorator()
+  async execute(command: SessionCommand): Promise<SessionResponseDto> {
+    const environment = await this.environmentRepository.findEnvironmentByIdentifier(command.applicationIdentifier);
+
+    if (!environment) {
+      throw new ApiException('Please provide a valid application identifier');
+    }
+
+    const inAppIntegration = await this.selectIntegration.execute(
+      SelectIntegrationCommand.create({
+        environmentId: environment._id,
+        organizationId: environment._organizationId,
+        userId: command.subscriberId,
+        channelType: ChannelTypeEnum.IN_APP,
+        providerId: InAppProviderIdEnum.Novu,
+        filterData: {},
+      })
+    );
+
+    if (!inAppIntegration) {
+      throw new NotFoundException('The active in-app integration could not be found');
+    }
+
+    if (inAppIntegration.credentials.hmac) {
+      validateHmacEncryption({
+        apiKey: environment.apiKeys[0].key,
+        subscriberId: command.subscriberId,
+        subscriberHash: command.subscriberHash,
+      });
+    }
+
+    const subscriber = await this.createSubscriber.execute(
+      CreateSubscriberCommand.create({
+        environmentId: environment._id,
+        organizationId: environment._organizationId,
+        subscriberId: command.subscriberId,
+      })
+    );
+
+    this.analyticsService.mixpanelTrack(AnalyticsEventsEnum.SESSION_INITIALIZED, '', {
+      _organization: environment._organizationId,
+      environmentName: environment.name,
+      _subscriber: subscriber._id,
+    });
+
+    const { count: unreadCount } = await this.notificationsCount.execute(
+      NotificationsCountCommand.create({
+        organizationId: environment._organizationId,
+        environmentId: environment._id,
+        subscriberId: command.subscriberId,
+        status: [MessagesStatusEnum.UNREAD],
+      })
+    );
+
+    const token = await this.authService.getSubscriberWidgetToken(subscriber);
+
+    return {
+      token,
+      unreadCount,
+    };
```

Comment:
In the new endpoint, we don't return subscriber data anymore, it's not needed. But we return the unread notifications count for the bell button.